### PR TITLE
Added copy as insert and copy as update

### LINKED ISSUE #649: feature request: Copy row result as sql
For example beeing able to right-click on a row and copy as UPDATE SQL or INSERT SQL would put that row in my clipboard as INSERT INTO X VALUES.....

### PR DESCRIPTION
**Pull request by jonaz**
Fixes #649

### CODE REVIEW

File: static/js/app.js [hunk 1]
```diff
@@ -1203,6 +1203,32 @@ function bindTableHeaderMenu() {
       var menuItem = $(e.target);
 
       switch(menuItem.data("action")) {
+        case "copy_as_insert":
```
Comment (sosedoff) [inline static/js/app.js:1206]: Can you wrap functionality from `copy_as_*` into a separate function to keep the handler function small?

File: static/js/app.js [hunk 2]
```diff
@@ -1225,6 +1251,38 @@ function bindTableHeaderMenu() {
   });
 }
 
+function getColumnsFromResults(){
+  let columns = [];
```
Comment (sosedoff) [inline static/js/app.js:1255]: Indentation on the following code is all messed up.

File: static/js/app.js [hunk 3]
```diff
@@ -1225,6 +1251,38 @@ function bindTableHeaderMenu() {
...
+    $("#results_header th").each(function(){
+    columns.push(this.innerText);
+    })
+  return columns;
+}
+
+function getValuesFromContext(context){
```
Comment (sosedoff) [inline static/js/app.js:1262]: Indentation on the following code is all messed up, please fix. Also remove the commented out line.

**Review by sosedoff — CHANGES_REQUESTED** [review]

**Comment by jonaz** [comment]
@sosedoff updated the commit with requested changes. Also make error when table is not selected. Could not find another way to figure out the table name.

**Comment by jonaz** [comment]
Ping @sosedoff ?

**Comment by sosedoff** [comment]
This looks good so far. One thing that may need to be addressed -- context menu does not work correctly when in query editor. Steps to reproduce:

1. Refresh pgweb page
2. Go to sql editor and run a query that returns some results
3. Right click on one of the rows - select copy as insert (will complain about table, thats expected)
4. Click on any table in the sidebar, it should render its contents
5. Navigate back to the sql editor, run the existing query again.
6. Right click on one of the rows - it will incorrectly copy insert statement using previously selected tables.

Id suggest we disable the context menu options on sql query results  page.

**Comment by a-patos** [comment]
Hello,
 is there any hope that MR will be successful one day or do we move on?

**Comment by sosedoff** [comment]
Closing for the time being. Im happy to revisit if OP is interested in continuing the work.

**Comment by jonaz** [comment]
Yes im interested @sosedoff

**Comment by jonaz** [comment]
New PR here https://github.com/sosedoff/pgweb/pull/730

**Comment by jonaz** [comment]
ping @sosedoff ?

**Pull request closed without merging**